src/linker/Linker/ConsoleLogger.cs

```diff
@@ -2,15 +2,59 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
+using System.Collections.Generic;
+using System.IO;
+using Mono.Cecil;
 
 namespace Mono.Linker
 {
 	public class ConsoleLogger : ILogger
 	{
-		public void LogMessage (MessageContainer message)
+		public Dictionary<MemberReference, string> ComputedStrings;
```

Comment:
Storing this per member reference will not always work. What if there are two warnings for the same member reference?
Anyway - the only reason to do is so that we could potentially sort the messages at the very end - but that won't work since the MemberRefereces will be useless at that point (if removed by Clean).
I would actually just immediately output all of the messages in the ProcessWarningsStep.

Thinking about this more - this would work with any logger - simply cache messages until the ProcessWarningStep - then potentially sort them and immediately output.

If there's any warning after that step, we can either cache and flush at the end , or disable the cache and immediately output.

That way we would get the same behavior for all loggers - which is probably preferable.

Reply:
> Simply cache messages until the ProcessWarningStep - then potentially sort them and immediately output.

I ended up doing this. We keep caching after the `ProcessWarningsStep` and then we flush at the very end.